src/librbd/mirror/DemoteRequest.h

```diff
@@ -18,12 +18,23 @@ namespace mirror {
 template <typename ImageCtxT = librbd::ImageCtx>
 class DemoteRequest {
 public:
+  static DemoteRequest *create(ImageCtxT &image_ctx, int64_t group_pool_id,
```

Comment:
I would recommend seeing if this can be done without passing the group specific information to what is an image specific operation. A better way might be to create a GroupDemoteRequest which would handle the group snapshot related operations.
ImageCtx has the GroupSpec information as well.